packages/crashlytics/ios/RNFBCrashlytics/RNFBCrashlyticsInitProvider.m

```diff
@@ -21,37 +21,54 @@
 #import "RNFBPreferences.h"
 #import "RNFBJSON.h"
 #import "RNFBMeta.h"
+#import "RNFBApp/RNFBSharedUtils.h"
 
 NSString *const KEY_CRASHLYTICS_DEBUG_ENABLED = @"crashlytics_debug_enabled";
 NSString *const KEY_CRASHLYTICS_AUTO_COLLECTION_ENABLED = @"crashlytics_auto_collection_enabled";
 
 @implementation RNFBCrashlyticsInitProvider
 
 + (void)load {
-  if ([self isCrashlyticsCollectionEnabled]) {
-    [FIRApp registerInternalLibrary:self withName:@"react-native-firebase-crashlytics" withVersion:@"6.0.0"];
-  }
+  [FIRApp registerInternalLibrary:self withName:@"react-native-firebase-crashlytics" withVersion:@"6.0.0"];
 }
 
 + (BOOL)isCrashlyticsCollectionEnabled {
   BOOL enabled;
 
   if ([[RNFBPreferences shared] contains:KEY_CRASHLYTICS_AUTO_COLLECTION_ENABLED]) {
     enabled = [[RNFBPreferences shared] getBooleanValue:KEY_CRASHLYTICS_AUTO_COLLECTION_ENABLED defaultValue:YES];
+    DLog(@"isCrashlyticsCollectionEnabled via RNFBPreferences: %d", enabled);
   } else if ([[RNFBJSON shared] contains:KEY_CRASHLYTICS_AUTO_COLLECTION_ENABLED]) {
     enabled = [[RNFBJSON shared] getBooleanValue:KEY_CRASHLYTICS_AUTO_COLLECTION_ENABLED defaultValue:YES];
+    DLog(@"isCrashlyticsCollectionEnabled via RNFBJSON: %d", enabled);
   } else {
+    // Note that if we're here, and the key is not set on the app's bundle, we default to "YES"
     enabled = [RNFBMeta getBooleanValue:KEY_CRASHLYTICS_AUTO_COLLECTION_ENABLED defaultValue:YES];
+    DLog(@"isCrashlyticsCollectionEnabled via RNFBMeta: %d", enabled);
   }
 
+  DLog(@"isCrashlyticsCollectionEnabled: %d", enabled);
+
   return enabled;
 }
 
 + (NSArray<FIRComponent *> *)componentsToRegister {
   return @[];
 }
 
+/*
+ * configureWithApp is automatically invoked by Firebase as this app is a registered FIRLibrary with the SDK.
+ *
+ * At this point "configure" has already been called on the FIRApp instance. This behavior is meant to mirror
+ * what is done in ReactNativeFirebaseCrashlyticsInitProvider.java
+ *
+ * This pattern may not be supported long term https://github.com/firebase/firebase-ios-sdk/issues/2933
```

Comment:
Very interesting callout here! I just scanned that issue and the resulting PRs

It looks like dynamic links has a very visible example of what the change means - that configureWithApp should go away, and that 'componentsToRegister' becomes the new natural home for the logic, with a new flag to initialize quickly

https://github.com/firebase/firebase-ios-sdk/pull/4137/files/71f589926b74e69ea76d0a25ae24e58efee6836b#diff-61367186bfcedbaaa5b0554cd03f8821R126-R162

What I don't know though is where to put that flag that says we want init eagerly (for the default app since crashlytics does not do multi-app) - e.g. this https://github.com/firebase/firebase-ios-sdk/pull/4137/files/71f589926b74e69ea76d0a25ae24e58efee6836b#diff-f33c561b2a3430b947e6da076758c6d9R1887

I think this might be good enough (it will certainly work now) but we need a separate issue to track this as they will appear set to purge it for v7